packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -163,33 +163,84 @@ export class SentryNestInstrumentation extends InstrumentationBase {
 
             target.prototype.intercept = new Proxy(target.prototype.intercept, {
               apply: (originalIntercept, thisArgIntercept, argsIntercept) => {
-                const [executionContext, next, args] = argsIntercept;
-                const prevSpan = getActiveSpan();
+                const context: MinimalNestJsExecutionContext = argsIntercept[0];
+                const next: CallHandler = argsIntercept[1];
+                const request = context.switchToHttp().getRequest();
 
-                return startSpanManual(getMiddlewareSpanOptions(target), (span: Span) => {
-                  const nextProxy = new Proxy(next, {
-                    get: (thisArgNext, property, receiver) => {
-                      if (property === 'handle') {
-                        const originalHandle = Reflect.get(thisArgNext, property, receiver);
-                        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                        return (...args: any[]) => {
-                          span.end();
-
-                          if (prevSpan) {
-                            return withActiveSpan(prevSpan, () => {
-                              return Reflect.apply(originalHandle, thisArgNext, args);
-                            });
-                          } else {
-                            return Reflect.apply(originalHandle, thisArgNext, args);
+                const parentSpan = getActiveSpan();
+                let afterSpan: Span;
+
+                return startSpanManual(getMiddlewareSpanOptions(target), (beforeSpan: Span) => {
+                  // eslint-disable-next-line @typescript-eslint/unbound-method
+                  next.handle = new Proxy(next.handle, {
+                    apply: (originalHandle, thisArgHandle, argsHandle) => {
+                      beforeSpan.end();
+
+                      if (parentSpan) {
+                        return withActiveSpan(parentSpan, () => {
+                          const handleReturnObservable = Reflect.apply(originalHandle, thisArgHandle, argsHandle);
+
+                          if (!request._sentryInterceptorInstrumented) {
+                            afterSpan = startInactiveSpan(
+                              getMiddlewareSpanOptions(target, 'Interceptor - After Route'),
```

Comment:
m: If we only create one span for multiple interceptors, I would make it plural:


```suggestion
                              getMiddlewareSpanOptions(target, 'Interceptors - After Route'),
```